### DISCUSSION TITLE
Linting a galaxy role's tests folder with symlink to role fails with newer version(s)

### DISCUSSION THREAD

**Discussion by chkpnt** [ansible/ansible-lint#4083] [Q&A] [answered]
I'm having issues to lint my role [`chkpnt.mailserver`](https://github.com/chkpnt/chkpnt-mailserver/tree/15.4.0) with the newest ansible-lint release 24.2.1. It worked fine with 6.10.0.

The issue is the directory structure: The roles for the tests are defined in `tests/testfixtures/roles/`, whereas `tests/testfixtures/roles/mailserver` is a symlink to the repository's root folder.

<img width="948" alt="grafik" src="https://github.com/ansible/ansible-lint/assets/1956979/b6750ea0-9a6b-449c-8232-656f3def21d5">

 I'm doing this, as I can't use `roles_path = testfixtures/roles:../` in [`tests/ansible.cfg`](https://github.com/chkpnt/chkpnt-mailserver/blob/15.4.0/tests/ansible.cfg#L3), as the galaxy role has to be in the repository's root instead of a folder named like the role's name.

With ansible-lint 6.10.0 this wasn't an issue:

![grafik](https://github.com/ansible/ansible-lint/assets/1956979/99b712ac-7fdf-4ae4-88bf-e9ccb4cd3814)

But with ansible-lint 24.2.1, the symlink is traversed multiple times, which leads to an OSError _File name too long_:

![grafik](https://github.com/ansible/ansible-lint/assets/1956979/b60a4a4d-09a9-494c-8356-31026bec9ab8)

So have anybody an idea for a work around?

**Comment by chkpnt** [comment, marked as answer]
Okay, I just figured it out, it's so obvious: After excluding the problematic path in `tests/.ansible-lint`, everything works as before. 

```yml
exclude_paths:
  - testfixtures/roles/mailserver
```

<img width="801" alt="grafik" src="https://github.com/ansible/ansible-lint/assets/1956979/ccb3fbe6-ad7a-4ba9-947f-c34377fa35e0">

**Comment by vincent-legoll** [comment]
I can add that exculsion from the command line (`--exclude some/recursive/path`) did not work for me, whereas your solution to exclude from the config file worked, thanks !
